Fix/feed parser

viacheslavbocharov opened the pull request:
fix(feed-parser): apply review fixes and stabilize
-Env: require DEFAULT_FEED_URL (URI) and add NODE_ENV; use app.config instead of process.env.
-Error handler: unified response; map 4xx/ENOTFOUND/timeouts; hide stack in prod.
-Parser: drop class; use functional makeUpstream4xx + isUpstream4xx.
-AutoLoad: load *.route.* from src/routes and src/modules only.
-Prisma plugin: add try/catch on $connect, logging, graceful onClose.
-Health: merge /health/server and /health/db into one routes file.
-Types/lint: remove non-null assertions, no any, small cleanups.

viacheslavbocharov committed:
fix(feed-parser): apply review fixes and stabilize

Env: require DEFAULT_FEED_URL (URI) and add NODE_ENV; use app.config instead of process.env.

Error handler: unified response; map 4xx/ENOTFOUND/timeouts; hide stack in prod.

Parser: drop class; use functional makeUpstream4xx + isUpstream4xx.

AutoLoad: load *.route.* from src/routes and src/modules only.

Prisma plugin: add try/catch on $connect, logging, graceful onClose.

Health: merge /health/server and /health/db into one routes file.

Types/lint: remove non-null assertions, no any, small cleanups.

viacheslavbocharov committed:
fix: biom linting

Gunnar97 commented on src/modules/feedParser/routes/feedParser.route.ts:
```diff
@@ -1,165 +1,67 @@
+        if (cache) {
+          const cachedItems = await getLatestItems(app.prisma, url, 50);
+          return reply.send({
+            sourceUrl: url,
```
Смотри, а мы не могли бы всю эту магию с кэшом, с кеш-айтемс, выполнять в сервисе с формированием объекта, а в Send, например, передавать только результат выполнения функции одной, чтобы это было все чище?

viacheslavbocharov replied:
Did route thiker. Moved logic to getFeedResponse.service.ts

Gunnar97 commented on src/plugins/prisma.plugin.ts:
```diff
@@ -2,13 +2,32 @@ import { PrismaClient } from "@prisma/client";
+      try {
+        await prisma.$disconnect();
+      } catch {}
+
```
Плохо оставлять кетч пустым.

viacheslavbocharov replied:
Fixed

Gunnar97 commented on src/routes/health.route.ts:
```diff
@@ -0,0 +1,19 @@
+      return { status: "ok" };
+    } catch (err) {
+      app.log.warn({ err }, "DB health failed");
+      return reply.status(503).send({ status: "fail" });
```
Уже юзался с sensible, а чем здесь роут отличается?

viacheslavbocharov replied:
Fixed, used @fastify/sensible

viacheslavbocharov committed:
fix: move logic to getFeedResponse.service.ts,  add error handlet to the empty catch, used @fastify/sensible

Gunnar97 commented on src/modules/feedParser/routes/feedParser.route.ts:
```diff
@@ -1,165 +1,35 @@
+
+  r.get("/feed", { schema }, async (req, reply) => {
+    try {
+      const q = req.query as { url?: string; force?: "0" | "1" };
```
типы должны тянуться из джейсон схемы, нет смысла создовать доп переменные сделай деструктуризацию req.query

viacheslavbocharov replied:
Fixed

Gunnar97 commented on src/modules/feedParser/routes/feedParser.route.ts:
```diff
@@ -1,165 +1,35 @@
+      return reply.send(data);
+    } catch (err) {
+      req.log.error({ err, query: req.query }, "feed route failed");
+      const e = err as { code?: string; message?: string };
```
фастифай лог

viacheslavbocharov replied:
Done

Gunnar97 commented on src/modules/feedParser/services/getFeedResponse.service.ts:
```diff
@@ -0,0 +1,44 @@
+
+export async function getFeedResponse(
+  app: FastifyInstance,
+  url: string,
```
тоже самое или в апп создаем апп или сюда принмаем фастифай

viacheslavbocharov replied:
Fixed

Gunnar97 commented on src/routes/health.route.ts:
```diff
@@ -0,0 +1,19 @@
+  app.get("/health/db", async (_req, reply) => {
+    try {
+      await app.prisma.feedCache.findFirst({ select: { url: true } });
+      return { status: "ok" };
```
есть просто пинг

viacheslavbocharov replied:
Fixed

viacheslavbocharov committed:
fix: /feed typing & logging, extract service, DB health ping

/feed: schema-typed req.query + req.log.error; delegate to getFeedResponse(app, url, force).

service: add getFeedResponse returning FeedResponse (cache + parse inside).

health: DB check via prisma.$runCommandRaw({ ping: 1 }), warn log, return 503 on failure.

schema/types: export FeedResponse and align response typing.
